mpc/README.md

```diff
@@ -156,7 +160,7 @@ specifying properties of the MPC:
 {
     "server": {
         "contributors_file": "contributors.json",
-        "start_time": "2019-10-02 17:00:00",   # Time (server-local)
+        "start_time": "2019-10-02 17:00:00",   # Time in UTC (server-local)
```

Comment:
Yes ofc...! I meant `UTC` here, not local. Will remove the `server-local`...

I think it makes sense to just consider all times to be UTC, the MPC is a global process (the coordinator can be in Central Europe - CEST) while a participant can be in the West coast of the US. UTC is the "common denominator", and does not require any time conversions, so it just looks to be the most suitable choice to handle time. Commands like `timedatectl` gives you the time UTC, so you just need to `timedatectl` and paste the UTC result in the server config file before you run the coordinator and start the server.

Reply:
Right, switching to UTC requires to use https://docs.python.org/3/library/time.html#time.gmtime instead of `localtime`

Reply:
> I think it makes sense to just consider all times to be UTC, the MPC is a global process (the coordinator can be in Central Europe - CEST) while a participant can be in the West coast of the US. UTC is the "common denominator", and does not require any time conversions, so it just looks to be the most suitable choice to handle time. Commands like timedatectl gives you the time UTC, so you just need to timedatectl and paste the UTC result in the server config file before you run the coordinator and start the server.

Yes, UTC is definitely the common zone and there are tools to convert.  My point was more that it might be easy for the coordinator admin to forget this step, and it can be very confusing to debug when this happens.  (So by requiring the zone to be specified we give a strict check against this - it's also a trivial code change.)

UTC-only will work of course, if you prefer that, but let's be careful to make it as clear as possible - including in the sample config files as well as this README.  I'd even consider changing the JSON attribute `start_time` to `start_time_utc` to make it absolutely clear, but perhaps that's overkill.

Reply:
Yes that a good point.

> I'd even consider changing the JSON attribute start_time to start_time_utc to make it absolutely clear, but perhaps that's overkill.

I think this is valid, I like the idea. It's more explicit and cannot hurt.

> UTC-only will work of course, if you prefer that

I would favor UTC only here indeed. It is easier to use UTC everywhere and easier to extend the current code base to have a better UX.
E.g. For now we do send an email to the contributor when his turn starts. A more user friendly piece of code may want - at the coordinator server's start time - to send an email to ALL contributors with an estimated starting time for their contribution window (so that they do not need to wait for their emails and can plan when - roughly - they'll need to be attentive, i.e. when they need to run `phaseX_contribute --wait-interval XX`). This can be done by carrying out some benchmark for all contribution steps (fetching challenge, computing, uploading response + coordinator's work between contributions -> all these steps are bandwidth/CPU dependent but benchmarks can be used to roughly know how big the contribution window needs to be). Afterwards, you multiply by the contributor's position in the pool + do some basic date arithmetic and you can send the approx contribution time in UTC to everyone up front.
That's just an example of additional step that can be done, and ofc you can do it using the server's time zone and then convert all results in UTC, but this is extra code that shouldn't need to be written IMO.

> My point was more that it might be easy for the coordinator admin to forget this step, and it can be very confusing to debug when this happens.

Indeed, I (almost) always use UTC by default - so I'd be confused if I needed to use the local time zone. Others may be confused the other way around. Good documentation + small changes in the code as you suggested (i.e. `start_time` -> `start_time_utc`) should be enough to carry the message out (and tbh I am yet to be convinced why a sysadmin managing a server communicating with other machines in arbitrary time zones would decide to use the local time zone on his server - even for logging activities - but I am not a sysadmin, so I may well be very wrong here!)

Reply:
#268 changes the time units to UTC only